src/datatable/datatable.jl

```diff
@@ -123,21 +132,21 @@ function DataTable(column_eltypes::Vector, cnames::Vector, nrows::Integer)
     p = length(column_eltypes)
     columns = Vector{Any}(p)
     for j in 1:p
-        columns[j] = NullableArray(column_eltypes[j], nrows)
+        columns[j] = Vector{column_eltypes[j]}(nrows)
     end
     return DataTable(columns, Index(cnames))
 end
 # Initialize an empty DataTable with specific eltypes and names
 # and whether a nominal array should be created
-function DataTable(column_eltypes::Vector{DataType}, cnames::Vector{Symbol},
+function DataTable(column_eltypes::Vector, cnames::Vector,
```

Comment:
This was a mistake. Not all of the constructors are as precise as this definition https://github.com/JuliaData/DataTables.jl/blob/master/src/datatable/datatable.jl#L122. I had modified all of them to be `column_eltypes::Vector{DataType}, cnames::Vector{Symbol}` and I reverted and accidentally changed this one during the revert. I'd like to specify the types for all of these constructors if we could add that to this PR.

And on this topic I noticed that indexing is done on Reals rather than Integers https://github.com/JuliaData/DataTables.jl/blob/20c71d6d40b3b238e902189b8262ba2b2e679b31/src/datatable/datatable.jl#L225.

That seems broader than it needs to be. Indexing with 1.0 and 1//1 both work.